src/Microsoft.DotNet.Tools.Run/Program.cs

```diff
@@ -19,52 +13,24 @@ public static int Main(string[] args)
         {
             DebugHelper.HandleDebugSwitch(ref args);
 
-            var app = new CommandLineApplication(throwOnUnexpectedArg: false);
-            app.Name = "dotnet run";
-            app.FullName = ".NET Executor";
-            app.Description = "Runner for the .NET Platform";
-            app.HelpOption("-h|--help");
-
-            var framework = app.Option("-f|--framework <FRAMEWORK>", "Compile a specific framework", CommandOptionType.MultipleValue);
-            var configuration = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
-            var preserveTemporaryOutput = app.Option("-t|--preserve-temporary", "Keep the output's temporary directory around", CommandOptionType.NoValue);
-
-            // This is required to be an option because otherwise we can't tell if the first argument is a project or the first argument to pass to an application
-            var project = app.Option("-p|--project <PROJECT_PATH>", "The path to the project to run (defaults to the current directory). Can be a path to a project.json or a project directory.", CommandOptionType.SingleValue);
+            RunCommand runCmd = new RunCommand();
 
-            app.OnExecute(() =>
+            ArgumentSyntax.Parse(args, syntax =>
             {
-                // Locate the project and get the name and full path
-                var path = project.Value();
-                if (!string.IsNullOrEmpty(path))
-                {
-                    if (File.Exists(path) && (Path.GetExtension(path) == ".csx"))
-                    {
-                        return RunInteractive(path);
-                    }
-                }
-                else
-                {
-                    path = Directory.GetCurrentDirectory();
-                }
-
-                var contexts = ProjectContext.CreateContextForEachFramework(path);
-                ProjectContext context;
-                if (!framework.HasValue())
-                {
-                    context = contexts.First();
-                }
-                else
-                {
-                    var fx = NuGetFramework.Parse(framework.Value());
-                    context = contexts.FirstOrDefault(c => c.TargetFramework.Equals(fx));
-                }
-                return Run(context, configuration.Value() ?? Constants.DefaultConfiguration, app.RemainingArguments, preserveTemporaryOutput.HasValue());
+                syntax.HandleErrors = false;
+                syntax.DefineOption("f|framework", ref runCmd.Framework, "Compile a specific framework");
```

Comment:
Oof, I definitely preferred the `var framework = ...` model of capturing these arguments over using `ref`, but I suppose if that's the direction `System.CommandLine` has gone...